### PR TITLE
Fix unit tests by using OS-chosen random port

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
retest this please

**Comment by tribal-tec** [comment]
retest this please

File: tests/connection/broker.cpp [hunk 1]
```diff
@@ -108,19 +108,23 @@ class Subscriber
...
+BOOST_AUTO_TEST_CASE( broker )
 {
+    // Using a different scheme so zeroconf resolution does not work
+    zeq::Publisher publisher( test::buildURI( "*" ));
+    _publisher = &publisher;
+    _broker.clear();
+
     Subscriber subscriber;
     std::thread thread( std::bind( &Subscriber::run, &subscriber ));
+    while( _broker.empty( ))
```
Comment (tribal-tec) [inline tests/connection/broker.cpp]: waitStarted?

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
retest this please

**Pull request merged**